x/gov/abci.go

```diff
@@ -57,15 +55,31 @@ func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
 		}
 
 		if passes {
-			handler := keeper.Router().GetRoute(proposal.ProposalRoute())
+			var (
+				messages []sdk.Msg
+				err      error
+				idx      int
+				msg      sdk.Msg
+			)
+
+			// attempt to execute all messages within the passed proposal
+			// Messages may mutate state thus we use a cached context. If one of
+			// the handlers fails, no state mutation is written and the error
+			// message is logged.
 			cacheCtx, writeCache := ctx.CacheContext()
+			messages, err = proposal.GetMsgs()
+			if err != nil {
+				for idx, msg = range messages {
+					handler := keeper.Router().Handler(msg)
+					_, err = handler(cacheCtx, msg)
+					if err != nil {
+						break
+					}
+				}
```

Comment:
this for block show go below of the `if err!=nil` block. We should just log smthg if `err!=nil`